pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,447 @@
+	// set default verbosity first
+	// it is used to fill the unattended verbosity with default verbosity
+	// key: JSONName, value: verbosity
+	var verbosityVal = map[string]uint{
+		"virtAPI":        virtconfig.DefaultVirtAPILogVerbosity,
+		"virtController": virtconfig.DefaultVirtControllerLogVerbosity,
+		"virtHandler":    virtconfig.DefaultVirtHandlerLogVerbosity,
+		"virtLauncher":   virtconfig.DefaultVirtLauncherLogVerbosity,
+		"virtOperator":   virtconfig.DefaultVirtOperatorLogVerbosity,
+	}
+
+	// if verbosity has been set in the KubeVirt CR, use the verbosity
+	lvMap, err := hasVerbosityInKV(kv)
+	if err != nil {
+		return nil, err
+	}
+	if len(lvMap) > 0 {
+		for key, value := range lvMap {
+			verbosityVal[key] = value
+		}
+	}
+
+	// create a message to show verbosity for the specified component
+	messages := createOutputMessage(verbosityVal, options)
+
+	return messages, nil
+}
+
+func setVerbosity(lvMap map[string]uint, options map[string]uint, patchData *[]patch.PatchOperation, op *string, path *string) {
+	// update lvMap based on the user-specified verbosity for all components
+	if verbosity, exist := options["all"]; exist {
+		for component := virtAPI; component < all; component++ {
+			JSONName := getJSONNameByVirtComponent(component)
+			lvMap[JSONName] = verbosity
+		}
+	}
+	// update lvMap based on the user-specified verbosity for each component
+	for componentName, verbosity := range options {
+		if componentName == "all" {
+			continue
+		}
+		JSONName := getJSONNameByComponentName(componentName)
+		lvMap[JSONName] = verbosity
+	}
+
+	if len(lvMap) != 0 {
+		addPatch(patchData, op, path, lvMap)
+	}
+}
+
+func addPatch(patchData *[]patch.PatchOperation, op *string, path *string, lvMap map[string]uint) {
+	*patchData = append(*patchData, patch.PatchOperation{
+		Op:    *op,
+		Path:  *path,
+		Value: lvMap,
+	})
+}
+
+func resetVerbosity(lvMap map[string]uint, patchData *[]patch.PatchOperation, op *string, path *string) {
+	// reset only if verbosity exists, otherwise do nothing
+	if len(lvMap) != 0 {
+		// add an empty object (removing the logVerbosity field can be another method)
+		emptyMap := map[string]uint{} // does not change the caller's lvMap
+		addPatch(patchData, op, path, emptyMap)
+	}
+}
+
+func createPatch(kv *v1.KubeVirt, options map[string]uint) ([]byte, error) {
+	patchData := []patch.PatchOperation{}
+	// just "add" is fine, no need of "replace" and "remove"
+	// https://www.rfc-editor.org/rfc/rfc6902
+	op := patch.PatchAddOp
+	path := "/spec/configuration/developerConfiguration/logVerbosity"
+
+	// if there is a logVerbosity field in the KubeVirt CR, fill in the data in the lvMap
+	lvMap, err := hasVerbosityInKV(kv)
+	if err != nil {
+		return nil, err
+	}
+	if lvMap == nil {
+		// if map is nil (logVerbosity field in the KubeVert CR is nil), need initialization
+		lvMap = make(map[string]uint)
+	}
+
+	if isReset {
+		resetVerbosity(lvMap, &patchData, &op, &path)
```

Comment:
path and op are always the same, no need to have them as arguments

Reply:
Agreed. This makes the code simpler. I left them as arguments if we need to use `remove` and/or `replace` operation. However, I should move them to constant now.